src/nvidia/src/kernel/gpu/rc/kernel_rc_notification.c

```diff
@@ -244,16 +244,17 @@ NV_STATUS krcErrorSetNotifier_IMPL
     KernelRc          *pKernelRc,
     KernelChannel     *pKernelChannel,
     NvU32              exceptType,
-    RM_ENGINE_TYPE     rmEngineType,
+    NvU32              type,
```

Comment:
Are there some updates possibly missing from the `nanovms/gpu-nvidia` repo in contrast to upstream Nvidia repo then?

Here https://github.com/nanovms/gpu-nvidia/blob/main/src/nvidia/generated/g_kernel_rc_nvoc.h#L382 it is `NvU32` and that's why I got the compilation error:
```
make -C src/nvidia
make[1]: Entering directory '/home/tuommaki/projects/nanovms.gpu-nvidia/src/nvidia'
 [ nvidia            ]  CC           src/kernel/gpu/rc/kernel_rc_notification.c
src/kernel/gpu/rc/kernel_rc_notification.c:241:11: error: conflicting types for ‘krcErrorSetNotifier_IMPL’ due to enum/integer mismatch; have ‘NV_STATUS(OBJGPU *, KernelRc *, KernelChannel *, NvU32,  RM_ENGINE_TYPE,  RC_NOTIFIER_SCOPE)’ {aka ‘unsigned int(struct OBJGPU *, KernelRc *, KernelChannel *, unsigned int,  RM_ENGINE_TYPE,  RC_NOTIFIER_SCOPE)’} [-Werror=enum-int-mismatch]
  241 | NV_STATUS krcErrorSetNotifier_IMPL
      |           ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from inc/kernel/gpu/rc/kernel_rc.h:2,
                 from src/kernel/gpu/rc/kernel_rc_notification.c:24:
generated/g_kernel_rc_nvoc.h:382:11: note: previous declaration of ‘krcErrorSetNotifier_IMPL’ with type ‘NV_STATUS(struct OBJGPU *, struct KernelRc *, struct KernelChannel *, NvU32,  NvU32,  RC_NOTIFIER_SCOPE)’ {aka ‘unsigned int(struct OBJGPU *, struct KernelRc *, struct KernelChannel *, unsigned int,  unsigned int,  RC_NOTIFIER_SCOPE)’}
  382 | NV_STATUS krcErrorSetNotifier_IMPL(struct OBJGPU *pGpu, struct KernelRc *pKernelRc, struct KernelChannel *pKernelChannel, NvU32 exceptType, NvU32 nv2080EngineType, RC_NOTIFIER_SCOPE scope);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1: all warnings being treated as errors
make[1]: *** [Makefile:224: _out/Nanos_x86_64/kernel_rc_notification.o] Error 1
make[1]: Leaving directory '/home/tuommaki/projects/nanovms.gpu-nvidia/src/nvidia'
make: *** [Makefile:35: src/nvidia/_out/Nanos_x86_64/nv-kernel.o] Error 2
```

Reply:
Our code is based on release 535.113.01 of the upstream Nvidia driver. Since then, there have been a couple more releases in the upstream repository, and if you look at https://github.com/NVIDIA/open-gpu-kernel-modules/blob/main/src/nvidia/generated/g_kernel_rc_nvoc.h you can see the function signature has been changed to use RM_ENGINE_TYPE as parameter type; more specifically, this change has been made in release 545.23.06.